pkg/primaza/controlplane/application_namespaces.go

```diff
@@ -88,7 +89,14 @@ func pushServiceBindingToNamespace(
 		},
 	}
 
+	l.Info("pushing service binding", "service-binding", sb.Name, "service-claim-status", sc.Status)
 	op, err := controllerutil.CreateOrUpdate(ctx, cli, &sb, func() error {
+		if sb.ObjectMeta.Annotations == nil {
+			sb.ObjectMeta.Annotations = map[string]string{}
+		}
+		sb.ObjectMeta.Annotations[constants.BoundRegisteredServiceNameAnnotation] = sc.Status.RegisteredService.Name
```

Comment:
Why did you pick adding a annotation instead part of the status object?

Reply:
as discussed let's approve this as is, and discuss it in community call